tests/test_other.py

```diff
@@ -2708,6 +2708,15 @@ def test_demangle_cpp(self):
     output = run_js('a.out.js')
     self.assertContained('Waka::f::a23412341234::point()', output)
 
+  # Test that malloc() -> OOM -> abort() -> stackTrace() -> jsStackTrace() -> demangleAll() -> demangle() -> malloc()
+  # cycle will not produce an infinite loop.
+  def test_demangle_malloc_infinite_loop_crash(self):
+    run_process([PYTHON, EMXX, path_from_root('tests', 'malloc_demangle_infinite_loop.cpp'), '-g', '-s', 'ABORTING_MALLOC=1', '-s', 'DEMANGLE_SUPPORT=1'])
+    output = run_js('a.out.js', assert_returncode=None, stderr=PIPE)
+    if output.count('Cannot enlarge memory arrays') > 2:
```

Comment:
I'm not sure what the constant 2 here, and below, mean? why isn't it just testing for the presence or lack of a message?

Reply:
Because the error message should appear, but not infinitely many times. In this test the error message will appear twice until demangle gets disabled.